proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -8,58 +8,47 @@
 
 ## Introduction
 
-Swift currently offers the ability for protocols that meet certain criteria to be used as types. Swift currently offers the ability for protocols with requirement meeting certain criteria to be used as types. Trying to use an unsupported protocol as a type yields the error: [the protocol] can only be used as a generic constraint because it has 'Self' or associated type requirements. This proposal aims to relax this artificial constraint, and instead constraint the availability of protocol members.
+Swift currently offers the ability for protocols that meet certain criteria to be used as types. Trying to use an unsupported protocol as a type yields the error: [the protocol] can only be used as a generic constraint because it has 'Self' or associated type requirements. This proposal aims to relax this artificial constraint, and instead constraint the availability of protocol members.
 
 ## Motivation
 
-In Swift, an existential type for a given protocol or protocol composition is a distinct type that can wrap a value of any conforming type. They are an incredibly useful feature in Swift for specific use cases that involve dynamicity. Being able to wrap different types that conform to a specified protocol, users and, more often, library authors can alleviate a lot of boilerplate code. That is, in certain use cases creating a custom existential type can be quite a daunting task due to its complexity; not to mention, that it introduces a lot of difficult-to-understand and hard-to-maintain code.
-
+In Swift, an *existential type* for a given *protocol type* or *protocol composition type* is a formally distinct type that has an equivalent spelling and can hold a value of any conforming type, only exposing the interface(s) of the specified protocol(s) (and superclass). The ability to represent an instance of any conforming type enables users and, more importantly, library authors to save on API surface and the considerable amount of boilerplate and difficult-to-maintain code often entailed in the design of ergonomic and exhaustive type-erased wrappers that stand in for a particular existential type. Likewise, existential types can prove incredibly useful in specific use cases that involve dynamicity.
 
 ### Heterogenous Collections
 
-Heterogeneous collections inherently involve some sort of dynamicity; for example, consider this `User` protocol:
+Heterogeneous collections inherently involve some sort of dynamicity. Consider this improvised `User` protocol:
 
 ```swift
 protocol User { 
-
   associatedtype ID : Hashable 
 
 
   var id: ID { get }
 
-  var username: String { get } 
-
+  var username: String { get }
 } 
 ```
 
 So, now we want to make business users and regular users that use different types of `ID`s:
 
 ```swift
 struct BussinessUser : User {
+  struct ID : Hashable {
```

Comment:
Did you mean to name `ID` `BusinessID` or did you forget to change the type name in the following declarations?

Reply:
I forgot to update the rest, thanks (it must be `ID` in both cases if we are to conform to `User`).